autoarray/plot/plotters/abstract_plotters.py

```diff
@@ -0,0 +1,299 @@
+from autoarray.plot.mat_wrap.wrap import wrap_base
+
+wrap_base.set_backend()
+
+import matplotlib.pyplot as plt
+
+from autoarray.plot.mat_wrap import visuals as vis
+from autoarray.plot.mat_wrap import include as inc
+from autoarray.plot.mat_wrap import mat_plot
+
+from functools import wraps
+from typing import Callable
+
+
+def title_from_func(func: Callable, index=None) -> str:
+    """If a title is not manually specified use the name of the function plotting the image to set the title.
+
+    Parameters
+    ----------
+    func : func
+       The function plotting the image.
+    """
+    if index is None:
+        return func.__name__.replace("figure_", "").capitalize()
+    return f"{func.__name__.replace('figure_', '').capitalize()}_{index}"
+
+
+def filename_from_func(func: Callable, index=None) -> str:
+    """If a filename is not manually specified use the name of the function plotting the image to set it.
+
+    Parameters
+    ----------
+    func : func
+       The function plotting the image.
+    """
+    funcname = func.__name__
+    filename = funcname.replace("figure_", "")
+
+    if index is None:
+        return filename
+    return f"{filename}_{index}"
+
+
+def index_from_inputs(args, kwargs):
+
+    try:
+        return args[1]
+    except IndexError:
+        key = next(iter(kwargs))
+        return kwargs[key]
+
+
+def for_figure(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+
+        if args[0].for_subplot:
+
+            args[0].set_mat_plots_titles_from_func(func=func)
+
+            return func(*args, **kwargs)
+
+        args[0].set_mat_plots_filenames_from_func(func=func)
+        args[0].set_mat_plots_titles_from_func(func=func)
+
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def for_figure_with_index(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+
+        index = index_from_inputs(args=args, kwargs=kwargs)
+
+        if args[0].for_subplot:
+
+            args[0].set_mat_plots_titles_from_func(func=func, index=index)

```

Comment:
If there will only ever be a single positional argument then give it an explicit name and don't use the * syntax

Reply:
Good shout.